### ISSUE TITLE
[Feature request] Implementing status of unsent and undelivered messages

### ISSUE THREAD

**Issue by elhananjair** [ukarim/smscsim#8]
Hello @ukarim 
Thank you for such nice tool, I was playing with smscsim and it works. But, for full simulation I want to request for feature which includes deliver_sm pdu response and show simulate unsent text message.

Thanks

**Comment by ukarim** [comment]
@elhananjair  just pushed new changes to this branch https://github.com/ukarim/smscsim/tree/issue8
Can you build from this branch and test it?
```
go build
export FAILED_DLRS=true
./smscsim
```
With this new env variable (FAILED_DLRS) every returned delivery report will have UNDELIVERABLE msg state

**Comment by elhananjair** [comment]
@ukarim Ok thanks I will do that, but what about submit_sm error responses, if command_status is not equal to 0.

**Comment by ukarim** [comment]
@elhananjair  made an update to the branch. Please look at this section in readme https://github.com/ukarim/smscsim/tree/issue8#env-variables

**Comment by ukarim** [comment]
@elhananjair  any updates on this?

**Comment by elhananjair** [comment]
I haven't tried it yet, I will let you know.

**Comment by elhananjair** [comment]
Hey @ukarim I couldn't build it with `go build export FAILED_DLRS=true ./smscsim` 

```

package export is not in GOROOT (/usr/src/export)
malformed import path "FAILED_DLRS=true": invalid char '='
```

**Comment by ukarim** [comment]
@elhananjair  this should be three separate commands
1 - go build
2 - export FAILED_SUBMITS=true
3 - ./smscsim

**Comment by ukarim** [comment]
@elhananjair  feature merged to master as is. If you want to test unsent messages, then you just need to set environment varaible `FAILED_SUBMITS=true`
Pushed changes to docker-hub. You can just do `docker run -e FAILED_SUBMITS=true -p 2775:2775 -p 12775:12775 ukarim/smscsim:0.2.0`
Thanks!

**Comment by elhananjair** [comment]
Hello, @ukarim thanks, I have tried it and it shows a successful status on sending the message and receiving delivery PDU. But I have got some comments:

1. `PDU.esm_class` for delivery should be equal to 4 as SMPP spec, but in this case, I am getting 0
2. The `PDU.source_addr` is empty as I logged its value, but it should be the destination number I have used in `submit_sm`

![image](https://github.com/ukarim/smscsim/assets/69236638/fe3e1d3c-c8f6-421e-bbef-6383fde1594a)


Thank you
